dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -704,25 +705,30 @@ protected void handlePublishUnpublish(String entryPath, String newName, boolean
                     publish(true, entryPath);
                 }
             } else if (!workflowExists(entryPath + "/" + newName)) {
-                try {
-                    // path should be represented as repository organization and name (ex. dockstore/dockstore-ui2)
-                    final Workflow newWorkflow = workflowsApi.manualRegister(
-                        getGitRegistry(existingWorkflow.getGitUrl()),
-                        existingWorkflow.getOrganization() + "/" + existingWorkflow.getRepository(),
-                        existingWorkflow.getWorkflowPath(),
-                        newName,
-                        existingWorkflow.getDescriptorType().toString(),
-                        existingWorkflow.getDefaultTestParameterFilePath()
-                    );
-
-                    final String completeEntryPath = entryPath + "/" + newName;
-
-                    out("Successfully registered " + completeEntryPath);
-
-                    workflowsApi.refresh(newWorkflow.getId(), true);
-                    publish(true, completeEntryPath);
-                } catch (ApiException ex) {
-                    exceptionMessage(ex, "Unable to publish " + entryPath + "/" + newName, Client.API_ERROR);
+                // Prevent specifying a custom name for both HOSTED and DOCKSTORE_YML workflows
+                if (Workflow.ModeEnum.HOSTED.equals(existingWorkflow.getMode()) || Workflow.ModeEnum.DOCKSTORE_YML.equals(existingWorkflow.getMode())) {
+                    out(WorkflowClient.INVALID_WORKFLOW_MODE_PUBLISH);
```

Comment:
Should be `err` not `out`? Or even `errorMessage`? Don't we want the program to exit with a non-zero exit code if it fails?